tree/tree/src/TBranchElement.cxx

```diff
@@ -2309,15 +2368,72 @@ Int_t TBranchElement::GetEntry(Long64_t entry, Int_t getall)
          case ROOT::kSTLunorderedmultimap:
             break;
          default:
+
             ValidateAddress(); // There is no ReadLeave for this node, so we need to do the validation here.
-            for (Int_t i = 0; i < nbranches; ++i) {
-               TBranch* branch = (TBranch*) fBranches.UncheckedAt(i);
-               Int_t nb = branch->GetEntry(entry, getall);
-               if (nb < 0) {
-                  return nb;
+            int nb(0);
+
+            auto seqProcessing = [&] () {
+               for (int i=0; i < nbranches; ++i) {
+                  auto branch = (TBranch*) fBranches.UncheckedAt(i);
+                  nb += branch->GetEntry(entry, getall);
                }
-               nbytes += nb;
+            };
+
+#ifdef R__USE_IMT
+            if (nbranches > 1 && ROOT::IsImplicitMTEnabled() && GetTree()->GetImplicitMT()) {
+               if (fSortedBranches.empty()) InitializeBranchLists();
```

Comment:
In conjunction with the return statement line 2265, does not that imply that InitializeBranchLists will be call at each invocation of TBranchElement::GetEntry? Is that the intent?